src/main/java/uk/gov/pay/adminusers/service/GovUkPayAgreementService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.pay.adminusers.service;
+
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import uk.gov.pay.adminusers.logger.PayLoggerFactory;
+import uk.gov.pay.adminusers.model.GovUkPayAgreement;
+import uk.gov.pay.adminusers.persistence.dao.GovUkPayAgreementDao;
+import uk.gov.pay.adminusers.persistence.entity.GovUkPayAgreementEntity;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class GovUkPayAgreementService {
+
+    private static Logger logger = PayLoggerFactory.getLogger(GovUkPayAgreementService.class);
+    private final GovUkPayAgreementDao agreementDao;
+    
+    @Inject
+    public GovUkPayAgreementService(GovUkPayAgreementDao govUkPayAgreementDao) {
+        this.agreementDao = govUkPayAgreementDao;
+    }
+
+
+    public Optional<GovUkPayAgreement> findGovUkPayAgreementByServiceId(int serviceId) {
+        return agreementDao.findByServiceId(serviceId)
+                .map((GovUkPayAgreementEntity::toGovUkPayAgreement));
+    }
+
+    public void doCreate(int serviceId, String email, LocalDateTime agreementTime) {
```

Comment:
I assume you’re following the convention from https://github.com/alphagov/pay-adminusers/pull/335 — unfortunately, that got this wrong: we want to use a `ZonedDateTime` (technically it could be an `Instant` but that would be inconsistent with what we do elsewhere).

(@danworth is going to fix it for PP-4539)